api4/post.go

```diff
@@ -69,6 +71,40 @@ func createPost(c *Context, w http.ResponseWriter, r *http.Request) {
 	w.Write([]byte(c.App.PostWithProxyAddedToImageURLs(rp).ToJson()))
 }
 
+func createEphemeralPost(c *Context, w http.ResponseWriter, r *http.Request) {
+	ephRequest := model.PostEphemeral{}
+
+	json.NewDecoder(r.Body).Decode(&ephRequest)
+	if ephRequest.UserID == "" {
+		c.SetInvalidParam("user_id")
+		return
+	}
+
+	if ephRequest.Post == nil {
+		c.SetInvalidParam("post")
+		return
+	}
+
+	ephRequest.Post.UserId = c.Session.UserId
+
+	if !c.App.SessionHasPermissionTo(c.Session, model.PERMISSION_CREATE_POST_EPHEMERAL) {
+		c.SetPermissionError(model.PERMISSION_CREATE_POST_EPHEMERAL)
+		return
+	}
+
+	if ephRequest.Post.CreateAt != 0 && !c.App.SessionHasPermissionTo(c.Session, model.PERMISSION_MANAGE_SYSTEM) {
+		ephRequest.Post.CreateAt = 0
```

Comment:
Why not set to `model.GetMillis()`?

Reply:
I've copy & pasted a major post of the handler itself, so I'd need to pass on that question and it's rather the question why https://github.com/DSchalla/mattermost-server/blob/6d85338436374fc80108209581e147b9f02bb87f/api4/post.go#L57 does it like that. Happy to change it of course!

Reply:
Ah, I see, I think that's correct so that the client will not be able to manipulate the `create_at` and the server will always set it to current time (or add 1 to current time in other case of ephemeral post).

Reply:
Actually, I think we should remove this if statement and set the time to `model.GetMillis()` like @saturninoabril said. This was included in the other handler because we a request to allow creating posts in the past but I don't think that makes any sense here since these are ephemeral